pyanaconda/configuration.py

```diff
@@ -0,0 +1,222 @@
+
+    def __init__(self, section_name, parser):
+        self._section_name = section_name
+        self._parser = parser
+
+    def _get_option(self, option_name, converter=None):
+        """Get a converted value of the option.
+
+        :param option_name: an option name
+        :param converter: a function or None
+        :return: a converted value
+        """
+        return get_option(self._parser, self._section_name, option_name, converter)
+
+    def _set_option(self, option_name, value):
+        """Set the option.
+
+        :param option_name: an option name
+        :param value: an option value
+        """
+        set_option(self._parser, self._section_name, option_name, value)
+
+
+class AnacondaSection(Section):
+    """The Anaconda section."""
+
+    @property
+    def debug(self):
+        """Run Anaconda in the debugging mode."""
+        return self._get_option("debug", bool)
+
+
+class ServicesSection(Section):
+    """The Services section."""
+
+    @property
+    def selinux(self):
+        """Enable SELinux usage in the installed system.
+
+        Valid values:
+
+         -1  The value is not set.
+          0  SELinux is disabled.
+          1  SELinux is enabled.
+        """
+        value = self._get_option("selinux", int)
+
+        if value not in (-1, 0, 1):
+            raise ValueError("Invalid value: {}".format(value))
+
+        return value
+
+
+class StorageSection(Section):
+    """The Storage section."""
+
+    @property
+    def arm_platform(self):
+        """The platform id of the ARM processor."""
+        return self._get_option("arm_platform", str)
+
+    @property
+    def dmraid(self):
+        """Enable dmraid usage during the installation."""
+        return self._get_option("dmraid", bool)
+
+    @property
+    def ibft(self):
+        """Enable iBFT usage during the installation."""
+        return self._get_option("ibft", bool)
+
+    @property
+    def gpt(self):
+        """Do you prefer creation of GPT disk labels?"""
+        return self._get_option("gpt", bool)
+
+    @property
+    def multipath_friendly_names(self):
+        """Use user friendly names for multipath devices."""
+        return self._get_option("multipath_friendly_names", bool)
+
+
+class AnacondaConfiguration(object):
+    """Representation of the Anaconda configuration."""
+
+    @classmethod
+    def from_defaults(cls):
+        """Get the default Anaconda configuration.
+
+        Read the current configuration from the temporary config file.
+        Or load the configuration from the configuration directory.
+
+        :return: an instance of AnacondaConfiguration
+        """
+        config = cls()
+
+        # Read the temporary configuration file.
+        config_path = os.environ.get("ANACONDA_CONFIG_TMP", ANACONDA_CONFIG_TMP)
+
+        # Or use the defaults if it doesn't exist.
+        if not config_path or not os.path.exists(config_path):
```

Comment:
It would be nice to log when the `ANACONDA_CONFIG_TMP` variable is set but the folder doesn't exists.

Reply:
Discussed, it is too early to log anything